# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1280511834 25200
#      Fri Jul 30 10:43:54 2010 -0700
# Node ID 6ea0cf31667d85cf5de3b8c1d913a3e6f424919f
# Parent  6abf966379dec3ad4180810e3adacd17d56cbbf5
Add Makefile options to clean up build directories and allow 'make distcheck' to finish

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2010-07-30  Rik <octave@nomad.inbox5.com>
+
+	* Makefile.am: add .gdbinit to the DISTCLEANFILES list
+
 2010-07-29  Jaroslav Hajek  <highegg@gmail.com>
 
 	* NEWS: Include list of new functions.
 
 2010-07-08  Rik <octave@nomad.inbox5.com>
 
 	* NEWS: Update with addition of engineering format.
 
diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -90,17 +90,17 @@ bin_SCRIPTS = \
   octave-bug$(SCRIPTS_EXE_SUFFIX)
 
 INFO_FILES = \
   INSTALL.OCTAVE \
   BUGS
 
 CLEANFILES = $(BUILT_SHELL_FILES)
 
-DISTCLEANFILES = $(INFO_FILES) 
+DISTCLEANFILES = $(INFO_FILES) .gdbinit 
 
 CONFIG_FILES = @ac_config_headers@ @ac_config_files@
 
 BUILT_SOURCES = run-octave
 
 nodist_octinclude_HEADERS = config.h
 
 all-local: $(bin_SCRIPTS) $(INFO_FILES) .gdbinit
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-07-30  Rik <octave@nomad.inbox5.com>
+
+	* Makefile.am: add .DOCSTRINGS to the DISTCLEANFILES list
+
 2010-07-29  Ben Abbott <bpabbott@mac.com>
 
 	* plot/print.m, plot/__print_parse_opts__.m, plot/__fltk_print__.m,
 	plot/__gnuplot_print__.m: Improved printing for the fltk backend.
 	* plot/private/__ghostscript__.m, plot/private/__tight_eps_bbox__.m:
 	New files to support printing.
 
 2010-07-29  Jaroslav Hajek  <highegg@gmail.com>
diff --git a/scripts/Makefile.am b/scripts/Makefile.am
--- a/scripts/Makefile.am
+++ b/scripts/Makefile.am
@@ -403,13 +403,14 @@ EXTRA_DIST += \
   $(FCN_FILES) \
   $(FCN_FILES_IN) \
   $(GEN_FCN_FILES) \
   DOCSTRINGS \
   mkdoc \
   mk-pkg-add
 
 DISTCLEANFILES = \
+  .DOCSTRINGS \
   DOCSTRINGS \
   $(PKG_ADD_FILES) \
   $(DIRSTAMP_FILES) \
   $(GEN_FCN_FILES)
 
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-07-30  Rik <octave@nomad.inbox5.com>
+
+	* Makefile.am: add .DOCSTRINGS to the DISTCLEANFILES list
+
 2010-07-30  John W. Eaton  <jwe@octave.org>
 
 	* pt-eval.cc (tree_evaluator::visit_if_command,
 	tree_evaluator::visit_switch_command): Also process breakpoints
 	for the command (bug #30490).
 
 2010-07-29  Ben Abbott <bpabbott@mac.com>
 
@@ -1418,17 +1422,17 @@ 2010-02-18  Thorsten Meyer  <thorsten.me
 
 	* dirfns.cc (Fglob): Document globbing patterns, add test.
 
 2010-02-18  Jaroslav Hajek  <highegg@gmail.com>
 
 	* load-path.cc (in_path_list): New helper function.
 	(add_to_fcn_map): Use it here.
 
-2010-02-18  Søren Hauberg  <hauberg@gmail.com>
+2010-02-18  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/__magick_read__.cc (__magick_finfo__):
 	Handle multiple frames in a single image file.
 
 2010-02-18  John W. Eaton  <jwe@octave.org>
 
 	* oct-map.h (Octave_map::Octave_map): Allow dimension to also be
 	specified in string_vector constructor.
@@ -7438,22 +7442,22 @@ 2008-12-25 Ben Abbott  <bpabbott@mac.com
 	* load-path.cc (Fcommandlinepath): New function.
 	* load-path.h (load_path::get_command_line_path,
 	load_path::do_get_command_line_path): New functions.
 
 2009-01-24 Ben Abbott  <bpabbott@mac.com>
 
 	* DLD-FUNCTIONS/eigs.cc: eigs.cc: skip tests if ARPACK is missing.
 
-2009-01-25  Søren Hauberg  <hauberg@gmail.com>
+2009-01-25  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* help.cc (do_get_help_text, raw_help_from_symbol_table): new output to
 	flag the a function is found but not documented.
 
-2009-01-25  Søren Hauberg  <hauberg@gmail.com>
+2009-01-25  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* help.cc (raw_help_from_file): No longer search for files called
 	'Contents.m', as this is moved to 'script/help.m'.
 
 
 2009-01-23  John W. Eaton  <jwe@octave.org>
 
 	* toplev.cc (octave_call_stack::do_goto_caller_frame):
@@ -7494,17 +7498,17 @@ 2009-01-22  John W. Eaton  <jwe@octave.o
 
 	* help.cc (do_which (std::ostream&, const std::string&), Fwhich):
 	Delete.
 	(do_which (const std::string&, std::string&), F__which__):
 	New functions.
 	* do_which (const std::string&):
 	Call do_which (const std::string&, std::string&) to do the work.
 
-2009-01-22  Søren Hauberg  <hauberg@gmail.com>
+2009-01-22  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* defun-int.h (print_usage): No longer mark as deprecated.
 	* defun.cc (print_usage): Simply call feval to execute print_usage.m.
 	* help.cc (additional_help_message, display_names_from_help_list,
 	display_symtab_names, simple_help, try_info, help_from_info,
 	display_help_text, display_usage_text, raw_help_from_list,
 	help_from_list, help_from_symbol_table, help_from_file,
 	builtin_help, Fhelp, display_file, do_type, Ftype,
@@ -7644,22 +7648,22 @@ 2009-01-14  Jaroslav Hajek  <highegg@gma
 	* oct-obj.cc (octave_value_list::normalize_null_values):
 	Rename to make_storable_values, use make_storable_value.
 	* oct-obj.h: Ditto.
 	* ov-builtin.cc: non_null_value -> storable_value.
 	* ov-cell.cc: Ditto.
 	* ov-struct.cc: Ditto.
 	* pt-decl.h: Ditto.
 
-2009-01-15  Søren Hauberg  <hauberg@gmail.com>
+2009-01-15  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/__magick_read__.cc (encode_uint_image):
 	Initialize bitdepth.
 
-2009-01-14  Søren Hauberg  <hauberg@gmail.com>
+2009-01-14  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/betainc.cc, DLD-FUNCTIONS/chol.cc,
 	DLD-FUNCTIONS/daspk.cc, DLD-FUNCTIONS/dasrt.cc,
 	DLD-FUNCTIONS/dassl.cc, DLD-FUNCTIONS/filter.cc,
 	DLD-FUNCTIONS/gammainc.cc, DLD-FUNCTIONS/gcd.cc,
 	DLD-FUNCTIONS/givens.cc, DLD-FUNCTIONS/hess.cc,
 	DLD-FUNCTIONS/lsode.cc, DLD-FUNCTIONS/qr.cc, DLD-FUNCTIONS/qz.cc,
 	DLD-FUNCTIONS/schur.cc, DLD-FUNCTIONS/svd.cc,
@@ -8621,17 +8625,17 @@ 2008-09-25  David Bateman  <dbateman@fre
 	concatenate is a class object, dispatch to the appropriate
 	vertcat/horzcat function.
 
 2008-09-25  John W. Eaton  <jwe@octave.org>
 
 	* symtab.cc (symbol_table::do_find): Don't set evaluated_args and
 	args_evaluated here, prior to call to symbol_table::fcn_info::find.
 
-2008-09-24  Søren Hauberg  <hauberg@gmail.com>
+2008-09-24  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/__magick_read__.cc (magick_to_octave_value): New
 	template function with specializations for various
 	GraphicsMagick++ types.
 	(F__magick_finfo__): New function.
 
 2008-09-24  John W. Eaton  <jwe@octave.org>
 
@@ -9065,17 +9069,17 @@ 2008-08-07  John W. Eaton  <jwe@octave.o
 	* ov.cc (octave_value::idx_type_value): Move definition here.
 	* ov.h: From here.
 
 	* DLD-FUNCTIONS/fsolve.cc (override_options): Don't fail if
 	options_map does not contain an expected keyword.
 	Fix typo in warning identifier.
 	(make_unimplemented_options): Use CamelCase names here.
 
-2008-08-06  Søren Hauberg  <hauberg@gmail.com>
+2008-08-06  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* error.cc (Ferror): Update format of error messages in exmple.
 	* parse.y: (Feval): Likewise.
 
 2008-08-06  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/fsolve.cc (hybrd_info_to_fsolve_info):
 	Update INFO values to be compatible with Matlab's current fsolve.
@@ -11657,17 +11661,17 @@ 2008-03-28  David Bateman  <dbateman@fre
 2008-03-27  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/max.cc: Rename from minmax.cc.
 	* Makefile.in (DLD_XSRC): Rename minmax.cc to max.cc.
 
 	* DLD-FUNCTIONS/__convn__.cc (convn): Use traits class and
 	typedefs to allow all types to be deduced from argument types.
 
-2008-03-27  Søren Hauberg  <hauberg@gmail.com>
+2008-03-27  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/__convn__.cc (Fconvn): Allow convolving real data with
 	complex data.
 
 2008-03-26  John W. Eaton  <jwe@octave.org>
 
 	* ov-range.h (octave_range::subsref (const std::string&,
 	const std::list<octave_value_list>&, int)):
@@ -11707,27 +11711,27 @@ 2008-03-26  John W. Eaton  <jwe@octave.o
 	is_constant method to decide which type of subsref method to call.
 	(octave_value::assign): Always call simple subref method here.
 
 	* DLD-FUNCTIONS/chol.cc, DLD-FUNCTIONS/det.cc,
 	DLD-FUNCTIONS/getpwent.cc, DLD-FUNCTIONS/inv.cc,
 	DLD-FUNCTIONS/qr.cc, DLD-FUNCTIONS/symrcm.cc, file-io.cc):
 	Texinfo fixes.
 
-2008-03-26  Søren Hauberg  <hauberg@gmail.com>
+2008-03-26  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/__convn__.cc (Fconvn):
 	Call complex_array_value to extract N-d array.
 
 2008-03-26  John W. Eaton  <jwe@octave.org>
 
 	* ov-base-sparse.cc (octave_base_sparse<T>::print_raw):
 	Also display percentage of elements that are nonzero.
 
-2008-03-25  Søren Hauberg  <hauberg@gmail.com>
+2008-03-25  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/__convn__.cc: New file.
 	* Makefile.in: Add __convn__.cc
 
 2008-03-25  David Bateman  <dbateman@free.fr>
 
 	* DLD-FUNCTIONS/hex2num.cc: New function
 	* Makefile.in (DLD_XSRC): Add hex2num.cc.
@@ -13997,17 +14001,17 @@ 2007-10-19  Olli Saarela  <Olli.Saarela@
 2007-10-17  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/spchol.cc (Fsymbfact): Delete special code for METIS.
 
 2007-10-17  Gabriele Pannocchia  <g.pannocchia@ing.unipi.it>
 
 	* DLD-FUNCTIONS/__qp__.cc (qp): Fix check for Wact(j).
 
-2007-10-15  Søren Hauberg  <hauberg@gmail.com>
+2007-10-15  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* error.cc (Ferror): Make text fit on pages when using smallbook.
 	* load-save.cc (Fsave_header_format_string): Ditto.
 	* ov-struct.cc (Fcell2struct): Ditto.
 	* DLD-FUNCTIONS/besselj.cc (Fairy): Ditto.
 	* DLD-FUNCTIONS/urlwrite.cc (Furlwrite, Furlread): Ditto.
 
 2007-10-15  David Bateman  <dbateman@free.fr>
@@ -14725,17 +14729,17 @@ 2007-06-25  Olaf Till  <i7tiol@t-online.
 	octave_steam_list::do_open_file_numbers,
 	octave_steam_list::do_get_file_number):
  	Use new octave_stream_list::list type.
 	(octave_stream_list::do_insert): Insert octave_stream with key
 	corresponding to file-descriptor.
 	(octave_stream_list::do_insert, octave_steam_list::insert):
 	Remove const qualifier of argument.
 
-2007-06-18  Søren Hauberg  <hauberg@gmail.com>
+2007-06-18  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* DLD-FUNCTIONS/__lin_interpn__.cc: Replace octave_NaN with octave_NA.
 
 2007-06-15  Shai Ayal  <shaiay@users.sourceforge.net>
 
 	* graphics.h (OCTAVE_GRAPHICS_PROPERTY_INTERNAL,
 	OCTAVE_GRAPHICS_PROPERTY, OCTAVE_GRAPHICS_MUTABLE_PROPERTY):
 	New macros. Use them to declare individual properties and define
@@ -14940,17 +14944,17 @@ 2007-05-22  John W. Eaton  <jwe@octave.o
 	(OCTAVE_RELEASE_DATE): Now 2007-05-22.
 
 2007-05-21  David Bateman  <dbateman@free.fr>
 
 	* debug.cc (Fdbstop): handle integer, vector and multiple line
 	arguments.
 	(Fdbclar): ditto. Eliminate extraneous debugging messages.
 
-2007-05-21  Søren Hauberg  <hauberg@gmail.com>
+2007-05-21  Sï¿½ren Hauberg  <hauberg@gmail.com>
 
 	* load-path.cc (Fpath, Frehash): Replace "LOADPATH" with "load
 	path" in doc strings.
 
 	* parse.y (Feval): Add example.
 
 2007-05-21  David Bateman  <dbateman@free.fr>
 
@@ -14961,26 +14965,26 @@ 2007-05-21  David Bateman  <dbateman@fre
 2007-05-21  John W. Eaton  <jwe@octave.org>
 
 	* oct-map.h (Octave_map::numel): Return number of array elements,
 	not number of map elements.
 	(Octave_map::nfields): Rename from length.
 	(Octave_map::empty): Delete.
 	Change all uses of empty to check nfields () == 0 instead.
 
-2007-05-21  Søren Hauberg  <soren@hauberg.org>
+2007-05-21  Sï¿½ren Hauberg  <soren@hauberg.org>
 
 	* help.cc (Fautoload): Doc fix.
 	* variables.cc (Fiscommand): Doc fix.
 
 2007-05-19  David Bateman  <dbatemna@free.fr>
 
 	* ov-fcn-inline.cc (Fvectorize): Doc fix.
 
-2007-05-16  Søren Hauberg  <soren@hauberg.org>
+2007-05-16  Sï¿½ren Hauberg  <soren@hauberg.org>
 
 	* ov.cc (Fsubsref, Fsubsasgn): Doc fix.
 
 2007-05-16  John W. Eaton  <jwe@octave.org>
 
 	* load-path.h (load_path::sys_path): New static data member.
 	(load_path::system_path): New static function.
 	(load_path::do_system_path): New function.
@@ -15038,24 +15042,24 @@ 2007-05-16  David Bateman  <dbateman@fre
 2007-05-14  Bob Weigel  <rweigel@gmu.edu>
 
 	* DLD-FUNCTIONS/svd.cc: Doc fix.
 
 2007-05-14  Thomas Weber  <thomas.weber.mail@gmail.com>
 
 	* DLD-FUNCTIONS/fft.cc (do_fft): Handle empty matrices.  New tests.
 
-2007-05-14  Søren Hauberg  <soren@hauberg.org>
+2007-05-14  Sï¿½ren Hauberg  <soren@hauberg.org>
 
 	* toplev.cc (Fatexit): Simplify example in doc string.
 	* help.cc (Flookfor): Doc fix.
 	* DLD-FUNCTIONS/cellfun.cc (Fcellfun):
 	Reformat to avoid long lines in doc string example.
 
-2007-05-13  Søren Hauberg  <soren@hauberg.org>
+2007-05-13  Sï¿½ren Hauberg  <soren@hauberg.org>
 
 	* toplev.cc (Fquit): Doc fix.
 	* help.cc (Fhelp): Doc fix.
 	* oct-hist.cc (Fsaving_history): Doc fix.
 
 2007-05-11  John W. Eaton  <jwe@octave.org>
 
 	* variables.cc (symbol_out_of_date):
@@ -15126,17 +15130,17 @@ 2007-04-27  David Bateman  <dbateman@fre
 	and mexSet.
 	* Makefile.in (INCLUDES): Add graphics.h
 
 2007-04-26  John W. Eaton  <jwe@octave.org>
 
 	* ov-usr-fcn.cc (octave_user_function::do_multi_index_op):
 	Only deal with varargout if ret_list->takes_varargs () is true.
 
-2007-04-26  Søren Hauberg  <soren@hauberg.org>
+2007-04-26  Sï¿½ren Hauberg  <soren@hauberg.org>
 
 	* DLD-FUNCTIONS/urlwrite.cc: Doc fix.
 
 2007-04-26  David Bateman  <dbateman@free.fr>
 
 	* pt-loop.cc (tree_simple_for_command::eval (void)): Correct
 	reshaping of dim_vector in for loop for multi-dimensional array.
 
@@ -15155,17 +15159,17 @@ 2007-04-23  Shai Ayal  <shaiay@users.sou
 	New arg A for alpha channel.  Set rgba instead of red, green, blue.
 	(color_property::color_property (char)): New constructor.
 	(color_propery::rgba): New data member.
 	(color_property::red, color_property::green, color_property::blue):
 	Delete.
 	(color_property::validate): Use rgba.
 	(color_property::c2rgba): New function.
 
-2007-04-23  Søren Hauberg  <soren@hauberg.org>
+2007-04-23  Sï¿½ren Hauberg  <soren@hauberg.org>
 
 	* data.cc (Fsize_equal): Allow more than two arguments.
 
 2007-04-20  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/__gnuplot_raw__.l (deftypefn):
 	(Vautomatic_replot): Delete static variable.
 	(Fautomatic_replot): Delete function.
@@ -16558,17 +16562,17 @@ 2006-09-15  John W. Eaton  <jwe@octave.o
 	* OPERATORS/op-double-conv.cc: Declare and install sq_string
 	conversions too.
 
 	* ov-base-int.cc (octave_base_int_scalar<T>::convert_to_str_internal,
 	octave_base_int_matrix<T>::convert_to_str_internal):
 	New functions.
 	* ov-base-int.h: Provide decls.
 
-2006-09-15  Søren Hauberg  <soren@hauberg.org>.
+2006-09-15  Sï¿½ren Hauberg  <soren@hauberg.org>.
 
 	* data.cc (Fsize): If nargout > ndims, fill with 1.
 
 2006-09-15  John W. Eaton  <jwe@octave.org>
 
 	* octave.cc: Fix xerbla decl.
 
 2006-09-13  John W. Eaton  <jwe@octave.org>
@@ -18664,17 +18668,17 @@ 2006-03-15  William Poetra Yoga Hadisoes
 
 2006-03-15  John W. Eaton  <jwe@octave.org>
 
 	* pager.cc (Fterminal_size): New function.
 
 	* help.cc (help_from_info): Simplify.
 	(try_info): Use feval to call doc instead of executing info program.
 	(additional_help_message): Point users to doc instead of help -i.
-	From Søren Hauberg <soren@hauberg.org>.
+	From Sï¿½ren Hauberg <soren@hauberg.org>.
 
 	* toplev.cc (Fsystem): Return output if nargout > 1, not 0.
 
 2006-03-14  Keith Goodman  <kwgoodman@gmail.com>
 
 	* help.cc (Fhelp, Fwhich, Flookfor): Doc string fix.
 
 2006-03-14  John W. Eaton  <jwe@octave.org>
@@ -19511,17 +19515,17 @@ 2005-08-25  David Bateman  <dbateman@fre
 
 	* OPERATORS/op-sm-m.cc, OPERATORS/op-sm-cm.cc, OPERATORS/op-scm-m.cc,
 	OPERATORS/op-scm-cm.cc, OPERATORS/op-m-sm.cc, OPERATORS/op-m-scm.cc,
 	OPERATORS/op-cm-sm.cc, OPERATORS/op-cm-scm.cc: Use mixed matrix/sparse
 	multiply operator rather than casting sparse to matrix.
 
 2005-07-18  John W. Eaton  <jwe@octave.org>
 
-	* strfns.cc (Fstrcmp): New function from Søren Hauberg
+	* strfns.cc (Fstrcmp): New function from Sï¿½ren Hauberg
 	<soren@hauberg.org> and Tom Holroyd <tomh@kurage.nimh.nih.gov>.
 	Adapt to Octave conventions.
 
 2005-07-11  David Bateman  <dbateman@free.fr>
 
 	* ov-fc-inline.cc (Fvectorize): Allow string arguments
 
 2005-07-08  John W. Eaton  <jwe@octave.org>
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -593,16 +593,17 @@ CLEANFILES = \
   doc-files \
   gendoc.cc \
   gendoc$(BUILD_EXEEXT) \
   graphics-props.cc \
   oct-parse.output \
   $(BUILT_NODISTFILES)
 
 DISTCLEANFILES = \
+  .DOCSTRINGS \
   DOCSTRINGS \
   $(OCT_FILES)
 
 MAINTAINERCLEANFILES = \
   $(BUILT_DISTFILES)
 
 octave_LDADD = $(OCTAVE_LIBS)
 
diff --git a/test/ChangeLog b/test/ChangeLog
--- a/test/ChangeLog
+++ b/test/ChangeLog
@@ -1,8 +1,13 @@
+2010-07-30  Rik <octave@nomad.inbox5.com>
+
+	* Makefile.am: New clean-local target to remove directories 
+	created automatically during testing.
+
 2010-04-28  John W. Eaton  <jwe@octave.org>
 
 	* test_struct.m: Update expected error message text.
 
 2010-03-19  Jaroslav Hajek  <highegg@gmail.com>
 
 	* test_slice.m: Use ID check rather than message for invalid resizing.
 
diff --git a/test/Makefile.am b/test/Makefile.am
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -76,13 +76,17 @@ EXTRA_DIST = \
   build_sparse_tests.sh \
   build_bc_overload_tests.sh \
   bc_overloads_expected \
   build_bc_overloads_expected.m \
   $(FCN_FILES)
 
 CLEANFILES = \
   test_sparse.m \
-  test_bc_overloads.m
+  test_bc_overloads.m \
+  tbcover.m
 
 DISTCLEANFILES = \
   fntests.log
-
+        
+clean-local:
+	-rm -rf @double @logical @int8 @uint64 @int64 @cell @single @int32 \
+	@uint8 @uint16 @uint32 @struct @int16 @function_handle @char
