# HG changeset patch
# User Carlo de Falco <kingcrimson@tiscali.it>
# Date 1331928889 -3600
#      Fri Mar 16 21:14:49 2012 +0100
# Node ID 045a5b85495d0ba5fec497f5c6b0305d94dd50ed
# Parent  d2c095e45196cf122bbd5667241ce2c83eaae805
maint: Remove redundant private function from the package manager.

* scripts/pkg/private/issuperuser.m: remove file.
* scripts/pkg/pkg.m: remove call to issuperuser.
* scripts/pkg/private/getarchprefix.m: remove call to issuperuser.
* scripts/pkg/private/create_pkgaddel.m: remove call to issuperuser.

diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -268,17 +268,19 @@ function [local_packages, global_package
   persistent local_list = tilde_expand (fullfile ("~", ".octave_packages"));
   persistent global_list = fullfile (OCTAVE_HOME (), "share", "octave",
                                      "octave_packages");
 
   confirm_recursive_rmdir (false, "local");
 
   mlock ();
 
-  global_install = issuperuser ();
+  ## If user is superuser set global_istall to true
+  ## FIXME: is it OK to set this always true on windows?
+  global_install = ((ispc () && ! isunix ()) || (geteuid () == 0));
 
   if (prefix == -1)
     if (global_install)
       prefix = fullfile (OCTAVE_HOME (), "share", "octave", "packages");
       archprefix = fullfile (octave_config_info ("libdir"),
                              "octave", "packages");
     else
       prefix = fullfile ("~", "octave");
diff --git a/scripts/pkg/private/getarchprefix.m b/scripts/pkg/private/getarchprefix.m
--- a/scripts/pkg/private/getarchprefix.m
+++ b/scripts/pkg/private/getarchprefix.m
@@ -19,16 +19,16 @@
 
 ## -*- texinfo -*-
 ## @deftypefn  {Function File} {[@var{archprefix} =}@
 ## getarchprefix (@var{desc}, @var{global_install})
 ## Undocumented internal function.
 ## @end deftypefn
 
 function archprefix = getarchprefix (desc, global_install)
-  if ((nargin == 2 && global_install) || (nargin < 2 && issuperuser ()))
+  if (global_install)
     archprefix = fullfile (octave_config_info ("libdir"), "octave",
-                           "packages", cstrcat(desc.name, "-", desc.version));
+                           "packages", cstrcat (desc.name, "-", desc.version));
   else
     archprefix = desc.dir;
   endif
 endfunction
 
diff --git a/scripts/pkg/private/issuperuser.m b/scripts/pkg/private/issuperuser.m
deleted file mode 100644
--- a/scripts/pkg/private/issuperuser.m
+++ /dev/null
@@ -1,32 +0,0 @@
-## Copyright (C) 2005-2012 Sï¿½ren Hauberg
-## Copyright (C) 2010 VZLU Prague, a.s.
-##
-## This file is part of Octave.
-##
-## Octave is free software; you can redistribute it and/or modify it
-## under the terms of the GNU General Public License as published by
-## the Free Software Foundation; either version 3 of the License, or (at
-## your option) any later version.
-##
-## Octave is distributed in the hope that it will be useful, but
-## WITHOUT ANY WARRANTY; without even the implied warranty of
-## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-## General Public License for more details.
-##
-## You should have received a copy of the GNU General Public License
-## along with Octave; see the file COPYING.  If not, see
-## <http://www.gnu.org/licenses/>.
-
-## -*- texinfo -*-
-## @deftypefn  {Function File} {@var{s} =} issuperuser ()
-## Undocumented internal function.
-## @end deftypefn
-
-function s = issuperuser ()
-  if ((ispc () && ! isunix ()) || (geteuid() == 0))
-    s = true;
-  else
-    s = false;
-  endif
-endfunction
-
